Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java

```diff
@@ -22,34 +22,114 @@
 
 package com.microsoft.azure.projectarcadia.common;
 
+import com.microsoft.azure.hdinsight.common.logger.ILogger;
 import com.microsoft.azure.hdinsight.sdk.cluster.SparkCluster;
 import com.microsoft.azure.hdinsight.spark.common.SparkSubmitStorageType;
 import com.microsoft.azure.hdinsight.spark.common.SparkSubmitStorageTypeOptionsForCluster;
+import com.microsoft.azure.hdinsight.sdk.rest.azure.projectarcadia.models.SparkCompute;
+import com.microsoft.azure.hdinsight.sdk.rest.azure.projectarcadia.models.SparkComputeProvisioningState;
 import com.microsoft.azuretools.authmanage.models.SubscriptionDetail;
 import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.client.utils.URIBuilder;
+
+import java.util.Optional;
+
+public class ArcadiaSparkCompute extends SparkCluster implements Comparable<ArcadiaSparkCompute>, ILogger {
+    public static final int ARCADIA_SPARK_SERVICE_PORT = 8660;
+    @NotNull
+    private ArcadiaWorkSpace workSpace;
+
+    @NotNull
+    private SparkCompute sparkComputeResponse;
+
+    public ArcadiaSparkCompute(@NotNull ArcadiaWorkSpace workSpace, @NotNull SparkCompute sparkComputeResponse) {
+        this.workSpace = workSpace;
+        this.sparkComputeResponse = sparkComputeResponse;
+    }
+
+    public boolean isRunning() {
+        if (getProvisioningState() == null) {
+            return false;
+        }
+
+        return getProvisioningState().equals(SparkComputeProvisioningState.PROVISIONING)
+                || getProvisioningState().equals(SparkComputeProvisioningState.SUCCEEDED);
+    }
+
+    @Nullable
+    public SparkComputeProvisioningState getProvisioningState() {
+        return this.sparkComputeResponse.provisioningState();
+    }
+
+    @NotNull
+    @Override
+    public String getState() {
+        return Optional.ofNullable(getProvisioningState()).map(state -> state.toString()).orElse("Unknown");
+    }
 
-public class ArcadiaSparkCompute extends SparkCluster {
     @NotNull
     @Override
     public String getName() {
-        return "";
+        return this.sparkComputeResponse.name();
     }
 
+    // This title is shown for spark compute list in run configuration dialog
     @NotNull
     @Override
     public String getTitle() {
-        return "";
+        if (getState().equalsIgnoreCase(SparkComputeProvisioningState.SUCCEEDED.toString())) {
```

Comment:
Can the `getState()` result to be parsed to `SparkComputeProvisioningState` before comparing?